trio/_core/_run.py

```diff
@@ -200,8 +200,7 @@ def collapse_exception_group(
         )
         return exceptions[0]
     elif modified:
-        # derive() returns Any for some reason.
-        return excgroup.derive(exceptions)  # type: ignore[no-any-return]
+        return excgroup.derive(exceptions)
```

Comment:
You might have forgotten to move this bit to the other PR?

Reply:
This is connected in a way yes, but the type ignore is not needed anymore because of updating exceptiongroup. The only mypy failures after this PR are the ones I intend to fix in #2787

Reply:
Ah sorry I assumed that other PR was for fixing type errors due to updating exceptiongroup instead of fixing that single one. Makes sense!